lib/ansible/plugins/action/__init__.py

```diff
@@ -650,9 +650,19 @@ def _update_module_args(self, module_name, module_args, task_vars):
         # make sure all commands use the designated shell executable
         module_args['_ansible_shell_executable'] = self._play_context.executable
 
-        # make sure all commands use the designated temporary directory
+        # make sure modules are aware if they need to keep the remote files
+        module_args['_ansible_keep_remote_files'] = C.DEFAULT_KEEP_REMOTE_FILES
+
+        # make sure all commands use the designated temporary directory if created
         module_args['_ansible_tmpdir'] = self._connection._shell.tmpdir
 
+        # make sure the remote_tmp value is sent through in case modules needs to create their own
+        try:
+            module_args['_ansible_remote_tmp'] = self._connection._shell.get_option('remote_tmp')
+        except KeyError:
+            # here for 3rd party shell plugin compatibility in case they do not define the remote_tmp option
+            module_args['_ansible_remote_tmp'] = '~/ansible'
```

Comment:
This seems inconsistent with other places (`~/.ansible/tmp`)

Reply:
Thanks, just pushed a change to use `~/.ansible/tmp` and also update the other location that had `~/ansible` set.